src/js/src/components/ExtractButton.jsx

```diff
@@ -7,10 +7,12 @@ import { makeOverpassAPIRequest } from '../actions.data';
 function ExtractButton({
     dispatch,
     disabled,
+    fetching,
 }) {
     const buttonClassName = disabled ?
         'button -extract -disabled' :
         'button -extract';
+    const text = fetching ? 'LOADING...' : 'EXTRACT';
```

Comment:
Probably not worth adjusting this but for a future enhancement we could split out the `fetching` state into `fetching data` and `creating shapefile` to have the indicator in the button be able to change contextually.